BeamDet/ERBeamDetSetup.cxx

```diff
@@ -769,13 +769,15 @@ void ERBeamDetSetup::ConstructGeometry() {
   // --------------------------------------------------------------------------
 }
 
-Double_t CalcElossIntegralVolStep (Double_t T, G4ParticleDefinition* ion, 
-                                   G4Material* mat, Double_t range) 
-{
+Double_t CalcElossIntegralVolStep (Double_t T, const G4ParticleDefinition* ion, 
+                                   const G4Material* mat, const Double_t range) {
+  assert(mat);
+  assert(ion);
+  if (range <= 0)
+    return 0.;
   Double_t integralEloss = 0.;
-  Double_t intStep = range / 5.;
+  const Double_t intStep = range / 5.;
```

Comment:
It is not your issue but in some reason range division in BeamDet reconstruction and QTelescope are different. In BeamDet it may be higher because inaccuracy in energy integration is less for incident ion as energy higher than products has. It needs to be checked.